graylog2-web-interface/src/views/components/SearchBar.tsx

```diff
@@ -95,7 +96,11 @@ const defaultOnSubmit = ({ timerange, streams, queryString }, currentQuery: Quer
     .query(createElasticsearchQueryString(queryString))
     .build();
 
-  return QueriesActions.update(newQuery.id, newQuery);
+  if (!currentQuery.equals(newQuery)) {
+    return QueriesActions.update(newQuery.id, newQuery);
```

Comment:
`QueriesActions.update` also contains a check if the query changed. Doing the check here as well still seems to be  a good solution, since we are using the action in other places as well